imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# imageio is distributed under the terms of the (new) BSD License.
+
+""" Read/Write images using rawpy.
+
+rawpy is an easy-to-use Python wrapper for the LibRaw library. 
+It also contains some extra functionality for finding and repairing hot/dead pixels.
+"""
+
+import rawpy
+import numpy as np
+
+from ..core.request import URI_BYTES, InitializationError, IOMode, Request
+from ..core.v3_plugin_api import PluginV3
+
+
+class RawPyPlugin(PluginV3):
+    """A class representing the rawpy plugin.
+    This class is a subclass of PluginV3.
+
+    Methods
+    -------
+
+    .. autosummary::
+    :toctree: _plugins/rawpy
+
+    RawPyPlugin.read
+    """
+    def __init__(self, request: Request) -> None:
+        """Instantiates a new rawpy plugin object
+
+        parameters
+        ----------
+        request: {Request}
+            A request object representing the resource to be operated on.
+        """
+
+        super().__init__(request)
+
+        self._image: rawpy = None
+        self._image_file = request.get_file()
```

Comment:
TODO: You are likely aware, but you will have to add some logic here that tests if rawpy can read the resource that is passed in. 

You'll have to verify the header, check the extension, or something similar. Likely rawpy has a built-in way of doing this and all you need to do is pass the file handle to rawpy, handle a potential exception, and raise `IOError` if the image is incompatible or can not be read with rawpy.

This is done because plugin selection will test a plugin's compatibility by instantiating it with the given request. If this works (no exception is raised) it will assume that the plugin can handle the request/image and will stop searching for alternatives.